antarest/launcher/business/slurm_launcher/slurm_launcher.py

```diff
@@ -0,0 +1,241 @@
+            time.sleep(2)
+
+    def start(self) -> None:
+        self.check_state = True
+        self.thread = threading.Thread(target=self._loop, daemon=True)
+        self.thread.start()
+
+    def stop(self) -> None:
+        self.check_state = False
+        self.thread = None
+
+    def add_callback(
+        self, callback: Callable[[str, JobStatus, bool], None]
+    ) -> None:
+        self.callbacks.append(callback)
+
+    def _init_launcher_arguments(self) -> argparse.Namespace:
+        main_options_parameters = MainOptionsParameters(
+            default_wait_time=self.config.launcher.slurm.default_wait_time,
+            default_time_limit=self.config.launcher.slurm.default_time_limit,
+            default_n_cpu=self.config.launcher.slurm.default_n_cpu,
+            studies_in_dir=str(
+                (
+                    Path(self.config.launcher.slurm.local_workspace)
+                    / "STUDIES_IN"
+                )
+            ),
+            log_dir=str(
+                (Path(self.config.launcher.slurm.local_workspace) / "LOGS")
+            ),
+            finished_dir=str(
+                (Path(self.config.launcher.slurm.local_workspace) / "OUTPUT")
+            ),
+            ssh_config_file_is_required=False,
+            ssh_configfile_path_prod_cwd=None,
+            ssh_configfile_path_prod_user=None,
+        )
+
+        parser: MainOptionParser = MainOptionParser(main_options_parameters)
+        parser.add_basic_arguments()
+        parser.add_advanced_arguments()
+        arguments = parser.parse_args([])
+
+        arguments.wait_mode = False
+        arguments.check_queue = False
+        arguments.json_ssh_config = None
+        arguments.job_id_to_kill = None
+        arguments.xpansion_mode = False
+        arguments.version = False
+        arguments.post_processing = False
+        return arguments
+
+    def _init_launcher_parameters(self) -> MainParameters:
+        main_parameters = MainParameters(
+            json_dir=self.config.launcher.slurm.local_workspace,
+            default_json_db_name=self.config.launcher.slurm.default_json_db_name,
+            slurm_script_path=self.config.launcher.slurm.slurm_script_path,
+            antares_versions_on_remote_server=self.config.launcher.slurm.antares_versions_on_remote_server,
+            default_ssh_dict_from_embedded_json={
+                "username": self.config.launcher.slurm.username,
+                "hostname": self.config.launcher.slurm.hostname,
+                "port": self.config.launcher.slurm.port,
+                "private_key_file": self.config.launcher.slurm.private_key_file,
+                "key_password": self.config.launcher.slurm.key_password,
+                "password": self.config.launcher.slurm.password,
+            },
+            db_primary_key="name",
+        )
+        return main_parameters
+
+    @staticmethod
+    def _delete_study(study_path: Path) -> None:
+        shutil.rmtree(study_path)
```

Comment:
what the point of wrapping this into a static method ? it obfuscate the fact that it's a potentially dangerous operation.
it should at least check that the path is below the antares workspace directory.